src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -857,7 +860,71 @@ where
         scope.clone(),
         move |mut _capabilities, _frontiers, scheduler| async move {
             let mut buffer = Vec::new();
+
+            let mut persist_clients = persist_clients.lock().await;
+            let persist_client = persist_clients
+                .open(storage_metadata.persist_location.clone())
+                .await
+                .expect("error creating persist client for Healthchecker");
+            drop(persist_clients);
+
+            let client_ref = &persist_client;
+
+            let write_to_persist = |row: Row, timestamp: Timestamp| async move {
+                let status_shard = match storage_metadata.status_shard {
+                    Some(s) => s,
+                    None => return,
+                };
+                let handle = client_ref
+                    .open_writer(status_shard)
+                    .await;
+
+                let mut handle = match handle {
+                    Ok(stuff) => stuff,
+                    Err(e) => {
+                        panic!("Invalid usage of the persist client for source {source_id} status history shard: {e:?}");
+                    }
+                };
+
+                let mut recent_upper = handle.upper().clone();
+                let mut append_ts = timestamp;
+                'retry_loop: loop {
+                    // We don't actually care so much about the timestamp we append at; it's best-effort.
+                    // Ensure that the append timestamp is past the current upper, and the new upper
```

Comment:
Is this one of these differential / timely "past"s?  I'm pretty tired at the moment but it looks like `append_ts` is not necessarily greater than `recent_upper` (current upper in the parlance of this comment)

Reply:
Right... maybe "not less than" is more precise?

Reply:
Yes, we should be able to write at the upper. Also, we ususally use "beyond" as meaning "not less than". I know... 😅